src/Microsoft.Azure.SignalR.Common/Endpoints/HubServiceEndpoint.cs

```diff
@@ -41,14 +43,24 @@ ServiceEndpoint endpoint
 
         public void CompleteScale()
         {
+            _pendingReload = false;
             _scaleTcs?.TrySetResult(true);
         }
 
+        // When remove an existing HubServiceEndpoint.
+        public void ResetScale()
+        {
+            _pendingReload = true;
+            _scaleTcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+        }
+
         public long UniqueIndex => _uniqueIndex;
 
         public override string ToString()
         {
             return base.ToString() + $"(hub={Hub})";
         }
+
+        internal override bool PendingReload => _pendingReload;
```

Comment:
Looks like `CompleteScale` and `ResetScale` can be merged into the `set` part of this property?

Reply:
Considering set false => complete task is a little hard to understand. Let's make it method easier to read.